typhos/utils.py

```diff
@@ -166,9 +166,24 @@ def random_color():
 
 
 class TyphosLoading(QtWidgets.QLabel):
+    LOADING_TIMEOUT_MS = 10000
     loading_gif = None
     """Simple widget that displays a loading GIF"""
-    def __init__(self, *args, **kwargs):
+    def __init__(self, enable_timeout=True, *args, **kwargs):
+        """
+        A QLabel with an animation for loading status.
+
+        Attributes
+        ----------
+        LOADING_TIMEOUT_MS : int
+            The timeout value in milliseconds for when to stop the animation
+            and replace it with a default timeout message.
+
+        Parameters
+        ----------
+        enable_timeout : bool
+            Whether or not to stop the animation after a timeout value.
+        """
```

Comment:
Usually we put these doc strings at the class level rather than attached to `__init__`. This should replace the `"""Simple widget that displays a loading GIF"""` text (which should have been at the very top of the class def)

Reply:
Addressed via https://github.com/pcdshub/typhos/pull/331/commits/68bb35867b474dd38f19bd8076de99987bffbaa9